app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -33,27 +33,37 @@ def proof(
           user_email: user_email,
         )
 
-        # todo(LG-8693): Begin verifying both the user's residential address and identity document
-        # address
+        residential_instant_verify_proof = proof_residential_address_if_needed(
```

Comment:
Hi @jmhooper, I wanted to seek out your opinion on how SOLID principles apply to this file.

When @night-jellyfish was synchronously reviewing this PR with me, she pointed out that the methods `proof_with_threatmetrix_if_needed`, `proof_residential_address_if_needed`, `proof_id_address_with_lexis_nexis_if_needed` and `proof_id_with_aamva_if_needed`, violate single responsibility principles because they A) determine if the proofing is necessary then B) do the actual proofing.  I saw that the `proof_with_threatmetrix_if_needed` method existed in resolution proofing job before Team Joy introduced progressive proofer and I wanted to seek out your thoughts on this pattern.  Thanks!